src/tests/functional/shared_test_classes/src/single_layer/memory.cpp

```diff
@@ -210,5 +213,9 @@ namespace LayerTestsDefinitions {
         }
     }
 
+    InferenceEngine::Blob::Ptr MemoryTest::GenerateInput(const InferenceEngine::InputInfo& info) const {
```

Comment:
Why do we need to override this in the generic class instead of in the GNA-specific file?

Reply:
put in other way: what's wrong with common set of values to be common for all plugins? if nothing, then this way we avoid burden of specialization.

Reply:
I guess that you're changing the way the function behaves for all plugins explicitly. And the way it's in the master, one change in the generic function will affect all places where it's used, without the need of further overriding.

Reply:
My approach is that to modify the shared classes you need to provide argumentation for this, why is it needed. Here we want to modify only GNA tests behavior, so I see no point in changing input generation for all tests using the shared tests class related to memory tests.

Reply:
I've changed the implementation to specialize GNA input and leave the comment why it was needed.